### PR TITLE
[New] Add react-datagrid@2.1.2 w/ npm auto-update

### LINKED ISSUE #9873: [Request] Add react-datagrid
**Library name:** react-datagrid
**Git repository url:**
**npm package url(optional):** 
**License(s):**
**Official homepage:**
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/40544647-request-add-react-datagrid?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by sufuf3**
close #9873, close #10546

Pull request for issue: #9873
Related issue(s): #11170  #10546

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/zippyui/react-datagrid (this link not found)
 * Official website (optional, not the repository): MIT
 * NPM package url (optional): https://www.npmjs.com/package/react-datagrid
 * License and its reference:
 * GitHub / Bitbucket popularity (required):
   - Count of watchers:
   - Count of stars:
   - Count of forks:
 * NPM download stats (optional):
   - Downloads in the last day: 11
   - Downloads in the last week: 1011
   - Downloads in the last month: 4603

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@dakshshah96  @cdnjs/intern3 
Please help me review this PR, Thank you.

Note:
The link of Git repository link (https://github.com/zippyui/react-datagrid) is disappear, but it's the necessary item. So I just follow the `package.json` from npm keep this item.

File: ajax/libs/react-datagrid/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "react-datagrid",
+  "version": "3.0.0",
+  "filename": "index.js",
+  "description": "React DataGrid",
```
Comment (dakshshah96) [inline ajax/libs/react-datagrid/package.json]: Let's make this more descriptive. Not necessary to follow `package.json` when they haven't provided a proper description, is it?

You can use `A carefully crafted DataGrid for React.` instead.

File: ajax/libs/react-datagrid/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "widget",
+    "table",
+    "list",
+    "react-component",
+    "react"
+  ],
+  "author": {
+    "name": "ZippyUI",
+    "email": "contact@zippyui.com"
+  },
```
Comment (dakshshah96) [inline ajax/libs/react-datagrid/package.json]: Please change author to `Radu Brehar` as given in the package's [package.json](https://unpkg.com/react-datagrid@3.0.0/package.json).

Comment (sufuf3) [inline ajax/libs/react-datagrid/package.json]: It's a little bit weird that if you `npm install react-datagrid` the `package.json` which under the `react-datagrid` folder, it show the author is ZippyUI.
I think that follow the origin source from npm which release by author zippyui is better than `Radu Brehar`.

File: ajax/libs/react-datagrid/package.json (hunk 1 above)
Comment (sufuf3) [inline ajax/libs/react-datagrid/package.json]: I'll update it. Thanks :+1:

File: ajax/libs/react-datagrid/package.json (hunk 2 above)
Comment (sufuf3) [inline ajax/libs/react-datagrid/package.json]: Oh, if you check this link https://unpkg.com/react-datagrid@2.0.0/package.json (use old version)
it shows the author is ZippyUI.

**Comment by sufuf3** [comment]
@dakshshah96 I modified the `description`  
Please help me review this PR, Thank you.

I didn't modify the author due to some consider (https://github.com/cdnjs/cdnjs/pull/11368#discussion_r120237199)

File: ajax/libs/react-datagrid/package.json (hunk 2 above)
Comment (dakshshah96) [inline ajax/libs/react-datagrid/package.json]: Alright, let's use ZippyUI 👍

**Comment by sufuf3** [comment]
@extend1994 I use `npm view react-datagrid versions` can get the v3.0.0.
Could you try again?

**Comment by sufuf3** [comment]
@cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

File: ajax/libs/react-datagrid/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "license": "MIT",
+  "npmName": "react-datagrid",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "!(react.|react-dom.)*.+(js|css|map)"
```
Comment (PeterDaveHello) [inline ajax/libs/react-datagrid/package.json]: Why so widely exclude `react.*` ? Shouldn't you just want to exlucde `react.min.js` & `react-dom.min.js`

**Comment by sufuf3** [comment]
I modified the Mapfiles rule.
@cdnjs/intern3 @dakshshah96 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@pvnr0082t @extend1994 @maruilian11 @qwe19272375 @hare1039 
Please help me review this PR, Thank you.

File: ajax/libs/react-datagrid/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "name": "ZippyUI",
+    "email": "contact@zippyui.com"
+  },
+  "license": "MIT",
+  "npmName": "react-datagrid",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "!(react.min|react-dom|react-dom.min).+(js|css|map)"
```
Comment (maruilian11) [inline ajax/libs/react-datagrid/package.json]: I think we should use `.gitignore` to ignore them, not use filemap. If you ignore the unstable versions, according to v3.0.0 file structure, we can still use `**/*` in dist.

Comment (sufuf3) [inline ajax/libs/react-datagrid/package.json]: I am a little bit confuse with your comment and https://github.com/cdnjs/cdnjs/pull/11368#discussion_r126990851 
I don't know which one is better? use `.gitignore` or modify `files` to exclude react.min.js & react-dom.min.js. 😞 
cc @PeterDaveHello

Comment (maruilian11) [inline ajax/libs/react-datagrid/package.json]: can be `react-datagrid/**/react.min.js` and `react-datagrid/**/react-dom*.js`

Comment (PeterDaveHello) [inline ajax/libs/react-datagrid/package.json]: I think both of them can work, auto-update config is a better way.

**Comment by sufuf3** [comment]
Just updated from `!(react.min|react-dom|react-dom.min).+(js|css|map)` to `**/!(react.min|react-dom|react-dom.min).+(js|css|map)`

@pvnr0082t @extend1994 @maruilian11 @qwe19272375 @hare1039
Please help me review this PR, Thank you.

File: ajax/libs/react-datagrid/package.json (hunk 4 above)
Comment (maruilian11) [inline ajax/libs/react-datagrid/package.json]: I mean we need to ignore `react-dom.min.js` and `react.min.js` of new versions not added yet.

File: ajax/libs/react-datagrid/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "name": "ZippyUI",
+    "email": "contact@zippyui.com"
+  },
+  "license": "MIT",
+  "npmName": "react-datagrid",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/!(react.min|react-dom|react-dom.min).+(js|css|map)"
```
Comment (maruilian11) [inline ajax/libs/react-datagrid/package.json]: `**/!(react.min*|react-dom*)` is enough

**Comment by sufuf3** [comment]
Modified the `files` part and update the .gitignore.
@maruilian11 
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@PeterDaveHello This lib repo not exist anymore, do we still need to put the repository info?

**Comment by PeterDaveHello** [comment]
npm is fine.

**Comment by maruilian11** [comment]
@PeterDaveHello Do you mean that we can change repository info to:
```
"repository": {
    "type": "npm",
    "url": "https://www.npmjs.com/package/react-datagrid"
  },
```

**Comment by PeterDaveHello** [comment]
npm is not a repository ... I just mean that we can still add it.

**Comment by sufuf3** [comment]
@pvnr0082t @maruilian11 @qwe19272375 @hare1039
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@PeterDaveHello I meet some problem during the rebasing. Please take a look.
Error msg:
```
"CDNJS_CACHE_USERNAME" secret not set!
```

**Comment by PeterDaveHello** [comment]
@maruilian11 It's not about rebasing.

**Comment by sufuf3** [comment]
@maruilian11 So is this PR fine?
Thank you.

**Comment by PeterDaveHello** [comment]
Can we simply exclude `index.js`, `react-dom.min.js` & `react.min.js`? The auto-update pattern looks odd and non-sense.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer I modified the pattern
And I think index.css is for index.js and index.html, so I exclude them.
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@sufuf3 Yes, `index.css` in dist before v2.1.2 is actually `normalize.css` v3.0.3, but `index.css` and `index.js` are different. Although they have been minified, after seeing their content, I think we still need to host them.

**Comment by sufuf3** [comment]
@maruilian11 I updated the pattern
Please help me review this PR, thanks.

**Comment by maruilian11** [comment]
@PeterDaveHello I see you mentioned we should exclude `index.js`, May I ask the reason? I did see some `reactdatagrid` keywords in the `index.js` in v3.0.0.

**Comment by PeterDaveHello** [comment]
Why do we need `index.js` just because of there is keyword in it? I just didn't see a reference says we need to use it, do we? Or don't we?

**Comment by sufuf3** [comment]
@cdnjs/intern3 please help me review this PR, thank you.

**Comment by maruilian11** [comment]
@PeterDaveHello Since the git repo has been removed, it's hard to know if `index.js` and `index.css` are the files we need. I think we can host it for some reasons:
1. They are under the `dist` and minified.
2. They are included in `index.html`
3. Before v3.0.0, the author(s) provide both `js` and `css`

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer I modified the pattern
to exclude index.js and index.html.
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/active-contributor @cdnjs/intern3 @cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@extend1994 but @PeterDaveHello said https://github.com/cdnjs/cdnjs/pull/11368#issuecomment-322531619 we don't need `index.js` and because this GitHub repo is disappeared so I can't discuss with the author.

1. Un, I don't think it really need to remove the cc, because https://libraries.io/github/zippyui-react show the author name is `ZippyUI` and the GitHub id is `zippyui-react`
2. I use npm install and the default version is v2.1.1, so I choose this version.

**Comment by extend1994** [comment]
I think we are discussing if we need index.js since #issuecomment-323253389. So I just say my opinion on it.

**Comment by sufuf3** [comment]
@PeterDaveHello could you give us some suggestion?
I don't think we need to add `index.js` because I didn't see any reference says we have to use it. 
And when I run this file, it's really different from `react-datagrid.js`.

But this PR seems stuffed in the `index.js` problem.

Thank you.

**Comment by PeterDaveHello** [comment]
What's the versions affected by this `index.js` issue?

**Comment by sufuf3** [comment]
@PeterDaveHello v3.0.0 ~ v3.0.2-beta47

**Comment by sufuf3** [comment]
@PeterDaveHello May I ask how can I continue to this PR?
Thank you.

**Comment by sufuf3** [comment]
Because reviewers want add `index.js` (https://github.com/cdnjs/cdnjs/pull/11368#pullrequestreview-59447849) but maintainer think we don't need to add it (https://github.com/cdnjs/cdnjs/pull/11368#issuecomment-322531619).
Also, there is no result about this.
Another hand, This lib's GitHub repo is disappear.

It's hard to discuss this issue with author.
I'll close this PR.
If there is a result for `index.js`, I'll reopen this PR and work on it.
Thanks.

**Pull request closed without merging**